### ISSUE TITLE
To Typescript or not to Typescript?

### ISSUE THREAD

**Issue by LBWright** [kentcdodds/ama#475]
I've been seeing a lot of articles about Typescript lately (thanks Google).
It's piqued my interest. I'm curious to know what you think about TS? Is it something I should concern myself with learning? Most of the articles have titles like "TS is the future of JS" (obviously clickbait, but hey, I'm not judging). I just wanted to know what you thought. 

I'm at an interesting point right now where I'm trying to figure out what to learn next. I'm a Frontend Developer primarily working with React and Modern Javascript. I'm essentially asking because I'm trying to figure out a next step for me in my dev knowledge. I'm worried that learning Typescript won't be profitable to me because it's not used with my current company. Or at least, when it comes to checking desired skills off a list, it doesn't have the highest level of profitability to learn next, if that makes sense. 

TLDR: Typescript - what do you think?

**Comment by kentcdodds** [comment]
Hi @LBWright!

My feelings about TS are similar to my feelings about Flow. Tacking static types on JavaScript is great, important, helpful, painful, and extremely frustrating. I never use typed JavaScript of any flavor in my open source projects because I feel like it would add too much of a barrier to contributors and it would get in my way developing the library. If I really want the guarantees of a typed language, then I should switch to [Reason](https://reasonml.github.io/). And I plan to do this for some things eventually.

Before Babel 7, my other oppositions to TypeScript were that I have to abandon my babel plugins that I find invaluable. But now I guess I can use them together without much trouble so that's ok I guess.

I think that you could be benefitted learning and using TypeScript. But I personally don't think I'll ever use it.

Good luck!
